eigensync/src/server/database.rs

```diff
@@ -0,0 +1,81 @@
+//! Server database layer for patch storage and peer management
+
+use crate::types::{Result, PeerId, ActorId};
+use rusqlite::Connection;
+use std::path::Path;
+
+/// Server database for storing patches and peer information
+pub struct ServerDatabase {
+    connection: Connection,
+}
+
+impl ServerDatabase {
+    /// Open or create a server database
+    pub async fn open<P: AsRef<Path>>(path: P) -> Result<Self> {
+        tracing::info!("Opening server database at {:?}", path.as_ref());
+        
+        // TODO: Implement actual database opening and migration
+        let connection = Connection::open(path)?;
+        
+        Ok(Self { connection })
+    }
+
+    /// Store a patch for a peer
+    pub async fn store_patch(
+        &self,
+        peer_id: PeerId,
+        actor_id: ActorId,
+        document_id: &str,
+        patch_data: &[u8],
```

Comment:
parse the patch before passing it to this function -> this function should take a `Change` type (I think, I don't think we're supposed to use `Patch`)